lib/modules/links/index.js

```diff
@@ -0,0 +1,100 @@
+import ModuleBase from './../../utils/ModuleBase';
+import { areObjectKeysContainedInOther } from './../../utils';
+
+const EVENT_TYPE = {
+  Link: 'dynamic_link_received',
+};
+
+/**
+ * @class Links
+ */
+export default class Links extends ModuleBase {
+  static _NAMESPACE = 'links';
+  static _NATIVE_MODULE = 'RNFirebaseLinks';
+
+  constructor(firebaseApp: Object, options: Object = {}) {
+    super(firebaseApp, options, true);
+  }
+
+  get EVENT_TYPE() {
+    return EVENT_TYPE;
+  }
+
+  /**
+   * Returns the link that triggered application open
+   * @returns {Promise.<String>}
+   */
+  getInitialLink() {
+    return this._native.getInitialLink();
+  }
+
+  /**
+   * Subscribe to dynamic links
+   * @param listener
+   * @returns {Function}
+   */
+  onLink(listener: Function): () => any {
+    const rnListener = this._eventEmitter.addListener(EVENT_TYPE.Link, listener);
+    return () => rnListener.remove();
+  }
+
+  /**
+   * Create long Dynamic Link from parameters
+   * @param parameters
+   * @returns {Promise.<String>}
+   */
+  createDynamicLink(parameters: Object = {}): Promise<String> {
+    if (!this._validateParameters(parameters)) {
+      return Promise.reject(new Error('Invalid Parameters.'));
+    }
+    return this._native.createDynamicLink(parameters);
+  }
+
+  /**
+   * Create short Dynamic Link from parameters
+   * @param parameters
+   * @returns {Promise.<String>}
+   */
+  createShortDynamicLink(parameters: Object = {}): Promise<String> {
+    if (!this._validateParameters(parameters)) {
+      return Promise.reject(new Error('Invalid Parameters.'));
+    }
+    return this._native.createShortDynamicLink(parameters);
+  }
+
+  _validateParameters(parameters: Object): boolean {
+    const suportedParametersObject = {
```

Comment:
Could this not be outside of the class?

Reply:
Also are all these parameters required?

Reply:
Yes, it could be outside of the class.
Just put it outside of the class on the same file?

In addition, these parameters are all supported. Not all of them required. I just validate that the input parameters object is contained in this object, to make sure there is no invalid parameters or parameters with typo.
The required parameters are checked in the native code.

Reply:
Aye just at the top outside class I guess, saves making a new instance of it every time.

And ok, makes sense :)